src/bvals/bvals_swarm.cpp

```diff
@@ -0,0 +1,166 @@
+//========================================================================================
+// Athena++ astrophysical MHD code
+// Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+//! \file bvals_swarm.cpp
+//  \brief functions that apply BCs for SWARMs
+
+#include "bvals/bvals_interfaces.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <cstring>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+
+#include "basic_types.hpp"
+#include "globals.hpp"
+#include "mesh/mesh.hpp"
+#include "parameter_input.hpp"
+#include "utils/buffer_utils.hpp"
+#include "utils/error_checking.hpp"
+
+namespace parthenon {
+
+BoundarySwarm::BoundarySwarm(std::weak_ptr<MeshBlock> pmb)
+    : bswarm_index(), pmy_block(pmb), pmy_mesh_(pmb.lock()->pmy_mesh) {
+#ifdef MPI_PARALLEL
+  swarm_id_ = pmb.lock()->pbval->bvars_next_phys_id_;
```

Comment:
Odd to access the member and not the function?

Reply:
Yes I agree... but this is what `CellCenteredBoundaryVariable` does: `cc_phys_id_ = pmb.lock()->pbval->bvars_next_phys_id_;`

Reply:
Yeah... this whole thing is odd. I think we keep what you did. But we really need to revisit and understand the boundary machinery.

Reply:
I'm unresolving until I had time to dig into the details.
Following our discussion about the Tags in our call last week, we need to be very careful with tags, so I'd prefer to sort this out here (not "fix/rewrite" it but to ensure that that we don't break code when, e.g., running multiple pieces (e.g., swarms and advection) with MPI at the same time.

Reply:
I agree @Pgrete about sorting this out now, I don't quite follow what tags are doing with `phys_id` but it seems like `Swarm`s and `Variable`s should both be using this to avoid the issue you point out. I'm curious to know what you think the path forward is, I'm happy to help figure this out.

Reply:
Right, it's currently unclear to me that this code actually can work unless you pack all of your variables to communicate over MPI into a single buffer (maybe this is how the code works now?)

I think it's OK if `Swarm`s and `Variable`s use the same `phys_id` counter, we just need to update that counter appropriately. What do you think the best path forward is, @pgrete? I could add another example that advects tracer particles with a velocity variable (so we communicate both particles and a CC field over MPI)?